src/testpool.jl

```diff
@@ -78,12 +76,13 @@ function replace_engine(name::String)
     end
```

Comment:
what does this function do? it seems like it just deletes and returns the next name, but maybe doesn't actually create that engine? - could you add a comment/docstring as appropriate to the function?

Reply:
The original intent was to allow replacement of a failed engine during testing. That's not been an issue, though, and the function isn't used.

I'm tempted to remove it, but if you ever did want to replace an engine the alternative is to delete and rebuild everything. I'll keep the method and update it to provision the engine.

Reply:
I fixed this to provision the engine, but I will remove it in a later PR when I change the threading model.